server/src/test/java/io/crate/metadata/settings/ValidatorsTest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.metadata.settings;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.elasticsearch.test.ESTestCase;
+import org.junit.Test;
+
+public class ValidatorsTest extends ESTestCase {
+
+    @Test
+    public void test_shows_trimmed_expected_values_in_error_messages() {
+        var validator = Validators.stringValidator("foo", "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k");
+        assertThatThrownBy(() -> validator.validate("nope"))
+            .hasMessage("Unsupported setting value: nope. Supported values are: a, b, c, d, e, f, g, h, i, j, ...");
```

Comment:
minor: Maybe you can test the exception class as well.

Reply:
It's not really relevant for the test scenario and already covered in another test so I'd rather leave it out. If for whatever reason we introduce a more specific exception there is then no need to adjust test cases like these.